src/concepts/kafka-proxies/mqtt-proxy.md

```diff
@@ -0,0 +1,114 @@
+---
+description: This guide will walk through the way Zilla manages MQTT Pub/Sub connections and messages.
+---
+
+# MQTT Kafka Proxy
+
+This guide will walk through the way Zilla manages MQTT Pub/Sub connections and messages.
+
+An MQTT server acts as a broker between publishers and subscribers. This requires a complex protocol to manage the wide range of IoT devices and use cases. By proxying these messages on and off of Kafka with the [mqtt-kafka](../../reference/config/bindings/binding-mqtt-kafka.md) binding, IoT devices can transmit data to a wider range of tech stacks, adapting to more business needs.
+
+Unlike other proxies, Zilla manages the different MQTT topics instead of blindly passing them down to Kafka. This way the Kafka architecture can be optimized for MQTT Pub/Sub. MQTT client subscribers and publishers will communicate with Zilla the same as any broker.
+
+## Step 1: Declaring the broker
+
+A Zilla MQTT server can manage client sessions and broker all of the messages sent.
+
+```yaml
+mqtt_server:
+  type: mqtt
+  kind: server
+  exit: mqtt_kafka_proxy
+
+mqtt_kafka_proxy:
+  type: mqtt-kafka
+  kind: proxy
+  options:
+    topics:
+      sessions: mqtt-sessions
+      messages: mqtt-messages
+      retained: mqtt-retained
+```
+
+### Protocol version
+
+The Zilla MQTT `server` supports the [MQTT v5.0 Specification].
+
+::: info Feature Coming Soon <HopeIcon icon="circle-right"/>
+[MQTT v3.1.1 Specification] support is currently on the [Zilla roadmap]. Star and watch the [Zilla repo] for new releases!
+:::
+
+[MQTT v5.0 Specification]:https://docs.oasis-open.org/mqtt/mqtt/v5.0/mqtt-v5.0.html
+[MQTT v3.1.1 Specification]:http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html
+[Zilla roadmap]:https://github.com/orgs/aklivity/projects/4
+[Zilla repo]:https://github.com/aklivity/zilla/releases
+
+### QOS
+
+The Zilla MQTT `server` supports the "At most once (QoS 0)" Quality of Service flag.
+
+::: info Feature Coming Soon <HopeIcon icon="circle-right"/>
+At least once (QoS 1) and Exactly once (QoS 2) delivery will be support is currently on the [Zilla roadmap]. Star and watch the [Zilla repo] for new releases!
+:::
+
+## Step 2: Pub/Sub message reflect with Kafka
+
+Zilla manages MQTT pub/sub using three kafka topics. The specific topic names can be configured using the [options.topics](../../reference/config/bindings/binding-mqtt-kafka.md#options-topics) property.
+
+```yaml
+topics:
+  messages: mqtt-messages
+  retained: mqtt-retained
+  sessions: mqtt-sessions
+```
+
+### Messages on Kafka
+
+All MQTT messages brokered by Zilla are published on the `messages` Kafka topic. The MQTT message topic becomes the kafka key
```

Comment:
This will change later when we support routing to different Kafka topics for different MQTT topic subtrees. (MQTT has a hierarchical topic space)

Reply:
I talked about that briefly with Barnabas. I figured once all of the future routing features are in place, I would update this page anyway and didn't want to worry about the semantics yet. 

Are you ok to leave it like this and updating it later?

Reply:
Yep this was just a heads up for you for later.